zproject/prod_settings_template.py

```diff
@@ -534,6 +535,26 @@
 ## "example.com"), otherwise leave this as None.
 # SSO_APPEND_DOMAIN = None
 
+## JWT-based users login and API keys fetching.
+##
+## If you want to login users or fetch API keys via JWT in Zulip,
+## you need to set the secret key and algorithm to use to validate
+## JWT tokens received.
+##
+## For more details see https://zulip.readthedocs.io/en/latest/production/authentication-methods.html#jwt
+# JWT_AUTH_KEYS: Dict[str, Any] = {
+#     # Subdomain for which you want to enable the service.
+#     "zulip": {
+#         # Shared secret key used to validate jwt tokens. This key
+#         # needs to be securely, randomly generated. Note that if you're
+#         # using the default HS256 algorithm, per RFC 7518, the key needs
+#         # to be at least 256 bits long.
+#         "key": "xxxxxx",
```

Comment:
Hmm, also thinking now that this should probably be moved to be read from `zulip-secrets.conf` instead?

Reply:
This would break backward compatibility, though, as the `JWT_AUTH_KEYS` dictionary was already in place and used by the `/json/fetch_api_key` API. Unless the upgrade scripts (e.g., `upgrade-zulip-from-git`) will take care of moving the key from `settings.py` to `zulip-secrets.conf`?

Reply:
Yes, that's correct.

Reply:
Changed this to use `get_secrets`.

@alexamirante This won't break compatibility, it'll only change where the key is fetched from for new installation. An old installation that has set this up will have their `/etc/zulip/settings.py` with `JWT_AUTH_KEYS` just specifying `key` there in the file, so it won't be affected. (That's because `prod_settings_template.py` simply becomes `/etc/zulip/settings.py` when you're doing a new installation)